tests/404_or_greater/expr_match.ml

```diff
@@ -0,0 +1,12 @@
+module M = struct
+  type t = Foo | Bar
+  type r = { i : int; t : t }
+end
+
+let f () =
+  let M.(s) = M.(Foo) in assert (s = M.Foo);
```

Comment:
I believe the only fully useful tests right now would be `match` expressions, where at least one of the cases has an or-pattern, i.e. `M.(Foo | Bar)`, `M.{i = 3 | 4; _}`, etc.:

- We aren't running the pattern rewriter on `let` expressions at all (#115).
- If the rewriter generates only one case (because there are no or-patterns), we throw the case away and don't generate a nested `match` expression (https://github.com/aantron/bisect_ppx/blob/d587132aa537da793a669612d7e3f9c0d3c57fa9/src/syntax/instrumentPpx.ml#L255).